common/app/routes/Settings/utils/legacyProjectData.js

```diff
@@ -0,0 +1,95 @@
+const legacyFrontEndProjects = {
+  challengeNameIdMap: {
+    'build-a-personal-portfolio-webpage': 'bd7158d8c242eddfaeb5bd13',
+    'build-a-random-quote-machine': 'bd7158d8c442eddfaeb5bd13',
+    'build-a-pomodoro-clock': 'bd7158d8c442eddfaeb5bd0f',
+    'build-a-javascript-calculator': 'bd7158d8c442eddfaeb5bd17',
+    'show-the-local-weather': 'bd7158d8c442eddfaeb5bd10',
+    'use-the-twitchtv-json-api': 'bd7158d8c442eddfaeb5bd1f',
+    'stylize-stories-on-camper-news': 'bd7158d8c442eddfaeb5bd18',
+    'build-a-wikipedia-viewer': 'bd7158d8c442eddfaeb5bd19',
+    'build-a-tic-tac-toe-game': 'bd7158d8c442eedfaeb5bd1c',
+    'build-a-simon-game': 'bd7158d8c442eddfaeb5bd1c'
+  },
+  challenges: [
+    'Build a Personal Portfolio Webpage',
+    'Build a Random Quote Machine',
+    'Build a Pomodoro Clock',
+    'Build a JavaScript Calculator',
+    'Show the Local Weather',
+    'Use the Twitchtv JSON API',
+    'Stylize Stories on Camper News',
+    'Build a Wikipedia Viewer',
+    'Build a Tic Tac Toe Game',
+    'Build a Simon Game'
+  ],
+  projectBlockName: 'Legacy Front End Projects',
+  superBlock: 'legacy-front-end'
+};
+
+const legacyBackEndProjects = {
+  challengeNameIdMap: {
+    'timestamp-microservice': 'bd7158d8c443edefaeb5bdef',
```

Comment:
Can you explain why this map is required? From past experience relying on challenge names are extremely brittle and leads to hard to track down bugs. Hard coding them like this will results in problems down the rode.